omnigibson/object_states/on_fire.py

```diff
@@ -0,0 +1,79 @@
+from omnigibson.macros import create_module_macros
+from omnigibson.object_states.link_based_state_mixin import LinkBasedStateMixin
+from omnigibson.object_states.object_state_base import AbsoluteObjectState
+from omnigibson.object_states.temperature import Temperature
+from omnigibson.object_states.heat_source_or_sink import HeatSourceOrSink
+
+
+# Create settings for this module
+m = create_module_macros(module_path=__file__)
+
+# TODO: Delete default values for this and make them required.
+m.DEFAULT_IGNITION_TEMPERATURE = 250
+m.DEFAULT_FIRE_TEMPERATURE = 1000
+m.DEFAULT_HEATING_RATE = 0.04
+m.DEFAULT_DISTANCE_THRESHOLD = 0.2
+
+
+class OnFire(HeatSourceOrSink):
+    """
+    This state indicates the heat source is currently on fire.
+
+    Once the temperature is above ignition_temperature, OnFire will become True and stay True.
+    Its temperature will further raise to fire_temperature, and start heating other objects around it.
+    It may include a heatsource_link annotation (e.g. candle wick), in which case the fire visualization will be placed
+    under that meta link. Otherwise (e.g. charcoal), the fire visualization will be placed under the root link.
+    """
+
+    def __init__(
+        self,
+        obj,
+        ignition_temperature=m.DEFAULT_IGNITION_TEMPERATURE,
+        fire_temperature=m.DEFAULT_FIRE_TEMPERATURE,
+        heating_rate=m.DEFAULT_HEATING_RATE,
+        distance_threshold=m.DEFAULT_DISTANCE_THRESHOLD,
+    ):
+        """
+        Args:
+            obj (StatefulObject): The object with the heat source ability.
+            ignition_temperature (float): The temperature threshold above which on fire will become true.
+            fire_temperature (float): The temperature of the fire (heat source) once on fire is true.
+            heating_rate (float): Fraction in [0, 1] of the temperature difference with the
+                heat source temperature should be received every step, per second.
+            distance_threshold (float): The distance threshold which an object needs
+                to be closer than in order to receive heat from this heat source.
+        """
+        assert fire_temperature > ignition_temperature, "fire temperature should be higher than ignition temperature."
+
+        super(OnFire, self).__init__(
```

Comment:
should just be `super().__init__(...)` just to be consistent with python3?

Reply:
done